src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -265,10 +266,97 @@ public void Stop()
+            }
+
+            if (_repl == null) {
+                StartInternal();
+            }
+
+            var result = new HashSet<string>();
+            C4Error err;
+            var ids = Native.c4repl_getPendingDocIDs(_repl, &err);
+
+            if (err.code > 0) {
+                WriteLog.To.Database.E(Tag, err.ToString());
+                throw CouchbaseException.Create(err);
+            }
+
+            if (ids != null) {
+                var flval = Native.FLValue_FromData(ids, FLTrust.Trusted);
+                var flarr = Native.FLValue_AsArray(flval);
+                var cnt = (int)Native.FLArray_Count(flarr);
+
+                for (int i = 0; i < cnt; i++) {
+                    var flv = Native.FLArray_Get(flarr, (uint)i);
+                    result.Add(Native.FLValue_AsString(flv));
+                    Native.FLValue_Release(flv);
+                }
+
+                Native.FLValue_Release(flval);
+            }
+
+            return result.ToImmutableHashSet<string>();
+        }
+
+        /// <summary>
+        /// Checks whether or not a document with the given ID has any pending revisions to push
+        /// </summary>
+        /// <param name="documentID">The document ID</param>
+        /// <returns>A bool which represents whether or not the document with the corresponding ID has one or more pending revisions.  
+        /// <c>true</c> means that one or more revisions have not been pushed to the remote yet, 
+        /// and <c>false</c> means that all revisions on the document have been pushed</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown if no push replication</exception>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        public bool IsDocumentPending([NotNull]string documentID)
+        {
+            CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(documentID), documentID);
+
+            if (!IsPushing()) {
+                WriteLog.To.Database.E(Tag, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
+                throw new CouchbaseLiteException(C4ErrorCode.Unsupported, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
+            }
+
+            if (_repl == null) {
+                StartInternal();
+            }
+
+            C4Error err;
+            var isDocPending = Native.c4repl_isDocumentPending(_repl, documentID, &err);
+
+            if (err.code > 0) {
+                WriteLog.To.Database.E(Tag, err.ToString());
+                throw CouchbaseException.Create(err);
+            }
+
+            return isDocPending;
+        }
+
         #endregion
 
         #region Private Methods
 
+        private bool IsPushing()
+        {
+            var push = Config.ReplicatorType.HasFlag(ReplicatorType.Push);
+            var pushnpull = Config.ReplicatorType.HasFlag(ReplicatorType.PushAndPull);
+            return push || pushnpull;
```

Comment:
`pushnpull` here is redundant.  The first line will return true for both cases (that's the meaning of "has flag")